registry/storage/garbagecollect.go

```diff
@@ -97,15 +98,32 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 			})
 		})
 
-		// In certain situations such as unfinished uploads, deleting all
-		// tags in S3 or removing the _manifests folder manually, this
-		// error may be of type PathNotFound.
-		//
-		// In these cases we can continue marking other manifests safely.
-		if _, ok := err.(driver.PathNotFoundError); ok {
-			return nil
+		if err != nil {
+			// In certain situations such as unfinished uploads, deleting all
+			// tags in S3 or removing the _manifests folder manually, this
+			// error may be of type PathNotFound.
+			//
+			// In these cases we can continue marking other manifests safely.
+			if _, ok := err.(driver.PathNotFoundError); !ok {
+				return err
+			}
+		}
+		blobService := repository.Blobs(ctx)
+		layerEnumerator, ok := blobService.(distribution.ManifestEnumerator)
+		if !ok {
+			return fmt.Errorf("unable to convert BlobService into ManifestEnumerator")
```

Comment:
if we are not formatting the error string, we should avoid using `fmt.Errorf` and simply go for `errors.New`